engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -890,4 +880,14 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
 
     runTransaction(action)
   }
+
+  private def materializeWorkflowDescriptor(id: WorkflowId, workflowSourceFiles: WorkflowSourceFiles)(implicit ec: ExecutionContext): Future[WorkflowDescriptor] = {
```

Comment:
Hi @gauravs90 - sorry for the delay, I was offsite all afternoon at a meeting. 

I chatted with @kshakir and it sounds like this is being done simply as a temporary measure because at the moment the widgets can't all be aligned properly and not viewed as the correct architecture. I'm fine with that, however longer term I still believe that the correct solution is to have a dedicated location in the actor system to provide validation services. 

I don't think we need to continue to block all of the work depending on this PR however and we can make a ticket to rearrange it down the road.